src/vm-vcpu/src/vcpu/mod.rs

```diff
@@ -61,6 +72,20 @@ pub struct VcpuState {
     pub cpuid: CpuId,
 }
 
+/// Represents the current run state of the VCPUs.
+#[derive(Default)]
+pub struct VcpuRunState {
```

Comment:
Not sure what the best state representation is (having a `VmRunState` in `VcpuRunState` feels a bit out of place) or whether a `Condvar` is the best primitive for communication, but we can definitely keep things like this for now and get a better idea as we start working on pause/resume and state save/restore. A couple of things to keep in mind are:

- Once operations such as pause/resume become available, we have to watch out for potential race conditions and state inconsistency (i.e. what happens when pause/unpause are called in random quick succession, given that communication with the vcpus isn't instant or deterministic).
- Data probably has to be passed around when saving state.

A potential alternative is to look at using [channels](https://doc.rust-lang.org/std/sync/mpsc/fn.channel.html) which provide synchronized communication and also allow sending data. This is just something to consider as we move forward.